### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by sdemarre**
I suspect this will need some improvements before it can be merged, one issue I see is that I don't have the texi index file, I don't know how to create that.
I'm happy to work on any other required changes or improvements.

### CODE REVIEW

File: sdemarre/diophantine_system/diophantine_system.asd [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	   :defsystem-depends-on ("maxima-file" "info-index")
+	   :name "diophantine-system"
+	   :maintainer "Serge De Marre"
+	   :author "Serge De Marre"
+	   :licence ""
+	   :description "diophantine system of linear equations solver"
+	   :long-description "Maxima program to solve a system of linear diophantine equations A.x=b with integer elements of A,x and b."
+
+	   :components
+	   ((:info-index "diophantine-system-index")
```
Comment (sdemarre) [inline sdemarre/diophantine_system/diophantine_system.asd]: this is a reference to a non existing file, is there some tool to generate this file from the texi file ?

**Comment by robert-dodier** [comment]
Hi Serge, thanks for working on this, looks terrific. About the Texinfo documentation, the index file is crerated by a script in maxima/doc/info which is called build_index.pl (it's a Perl script) which takes the .info file as input and prints the index to standard output. I have just now tried it and it works as expected after fixing 2 small problems in the .texi file (I'll attach a patch for that):

```
$ makeinfo diophantine_system.texi
$ perl path/to/maxima/doc/info/build_index.pl diophantine_system.info > diophantine_system-index.lisp
```

Now in Maxima if I say `load("diophantine_system-index.lisp")`, documentation search `?` and `??` will find the new items.

I've attached a patch to fix the .texi problems. (1) missing @end example, (2) fix up node references.

best,
Robert

[diophantine_system-patch.txt](https://github.com/maxima-project-on-github/maxima-packages/files/4302315/diophantine_system-patch.txt)

**Comment by sdemarre** [comment]
Hello Robert,
I applied your patch, added the generated files (info and lisp index) and fixed a wrong reference in the asd file. 
I'm not sure the asd file is useful though, since the current maxima-asdf would not be able to fetch a single package (e.g. diophantine_system) the way we have them organized right now (correct me if I'm wrong).

**Comment by robert-dodier** [comment]
Hi Serge, about the ASD file, I guess I am inclined to just let it stand. Even if it doesn't work right, we'll fix it up at some point, I'm not too worried about it.

**Comment by sdemarre** [comment]
Ok, then we are good to merge, as far as I'm concerned.

**Pull request merged**